### ISSUE TITLE
`object_usage_linter` false positive: Variable assigned **and used** within `with` flagged as unused

### ISSUE THREAD

**Issue by katrinabrock** [jimhester/lintr#3020]
### MRE

```r
lint(
  "
  fn <- function(df) {
    with(df, {
      x <- b + c
      x
    })
  }",
  linters = list(object_usage_linter(skip_with = TRUE))
)
```

### Expected Behavior

No lint. `x` should not be considered as "assigned and not used" for two reasons: 1) It's inside a `with` statement 2) it is being used! In fact, it's the return value of the function as a whole.

### Actual Behavior
```
<text>:4:7: warning: [object_usage_linter] local variable 'x' assigned but may not be used
      x <- b + c
```

Lintr version: `lintr_3.3.0-1`
